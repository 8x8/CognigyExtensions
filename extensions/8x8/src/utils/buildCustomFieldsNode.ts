import type { INodeField } from '@cognigy/extension-tools/build/interfaces/descriptor';
import { addFilterKeyPrefix } from './filterPrefix';

const buildCustomFieldsNode = (): INodeField => ({
  key: addFilterKeyPrefix('customFields'),
  label: 'Custom Fields',
  type: 'json',
  description: 'Custom fields and their parameter names are added by the administrator of the tenant.',
  defaultValue: {}
});

export default buildCustomFieldsNode;
