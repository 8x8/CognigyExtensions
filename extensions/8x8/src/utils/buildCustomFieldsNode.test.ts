import buildCustomFieldsNode from './buildCustomFieldsNode';

describe('utils > buildCustomFieldsNode', () => {
  it('should create the nodes for customer logic', () => {
    const node = buildCustomFieldsNode();

    expect(node).toMatchSnapshot();
  });
});
