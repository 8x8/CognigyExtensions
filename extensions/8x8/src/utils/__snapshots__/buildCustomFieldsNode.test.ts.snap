// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils > buildCustomFieldsNode should create the nodes for customer logic 1`] = `
{
  "defaultValue": {},
  "description": "Custom fields and their parameter names are added by the administrator of the tenant.",
  "key": "filter$customFields",
  "label": "Custom Fields",
  "type": "json",
}
`;
